doc/developers/maintainer.rst

```diff
@@ -22,18 +47,26 @@ For more information see https://github.com/scikit-learn/scikit-learn/wiki/How-t
 
         $ git shortlog -ns 0.998..
 
+   - Update the release date in whats_new.rst
+
    - Edit the doc/index.rst to change the 'News' entry of the front page.
 
-2. Update the version number in sklearn/__init__.py, the __version__
-   variable
+   - Note that these changes should be made in master and cherry-picked into
+     the release branch.
+
+2. On the branch for releasing, update the version number in
+   sklearn/__init__.py, the ``__version__`` variable by removing ``dev*`` only
+   when ready to release.
+   On master, increment the verson in the same place (when branching for
+   release).
 
 3. Create the tag and push it::
 
-    $ git tag 0.999
+    $ git tag -a 0.999
 
-    $ git push origin --tags
+    $ git push https://github.com/scikit-learn/scikit-learn --tags
```

Comment:
probably `upstream` would make more sense. Or the `git-ssh` link instead of `https`, since `https` and 2FA are not besties. On the other hand, at the point somebody's doing this release, they can figure this out themselves. So no hard feelings.

Reply:
https and 2FA are fine as long as you keep your token. Shrug. Can change